src/Cake.Scripting/CodeGen/IScriptAliasFinder.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Cake.Core.IO;
+
+namespace Cake.Scripting.CodeGen
+{
+    public interface IScriptAliasFinder
+    {
+        IReadOnlyList<CakeScriptAlias> FindAliases(FilePath path);
```

Comment:
IEnumerable<T> might have side effects (implementation might be a enumerator state machine instead of a class just implementing it), so I prefer to be more strict with return values.